UniAtHome/UniAtHome.WebAPI/Middleware/ErrorHandlingMiddleware.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Hosting;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace UniAtHome.WebAPI.Middleware
+{
+    public class ErrorHandlingMiddleware
+    {
+        private readonly RequestDelegate next;
+
+        public ErrorHandlingMiddleware(RequestDelegate next)
+        {
+            this.next = next;
+        }
+
+        public async Task Invoke(HttpContext context, IWebHostEnvironment env)
+        {
+            try
+            {
+                await this.next(context);
+            }
+            catch (Exception ex)
+            {
+                await HandleExceptionAsync(context, ex, env);
+            }
+        }
+
+        private static Task HandleExceptionAsync(HttpContext context, Exception exception, IWebHostEnvironment env)
+        {
+            HttpStatusCode status;
+            string message;
```

Comment:
Why do you declare and initialize variables on the separate lines?

Reply:
Because later we will handle some other exceptions, and all exceptions are handled in different ways, with different messages and status codes (in other words, initialized differently).

Reply:
IMHO, it's better not to rewrite this method every time we add new exceptions. I've done this before using polymorphism ([here is an example](https://github.com/EdShel/ESportRaise/blob/2d3648823d15050d31602f10332301d12b51f23f/BackEnd/ESportRaise.BackEnd/ESportRaise.BackEnd.API/Middleware/ExceptionMiddleware.cs#L19)). Personally, I don't think it's critical today, but if one day we decide to catch not only server errors (500 code), it'd be a pain the ass

Reply:
i think it'll be useful for the most common and frequent exceptions, so that we won't make our code messy with huge amount of try/catch blocks